components/admin_console/team_channel_settings/channel/list/channel_list.tsx

```diff
@@ -15,25 +17,36 @@ import SearchIcon from 'components/widgets/icons/search_icon';
 
 import ChannelRow from './channel_row';
 
-export default class ChannelList extends React.PureComponent {
-    static propTypes = {
-        actions: PropTypes.shape({
-            searchAllChannels: PropTypes.func.isRequired,
-            getData: PropTypes.func.isRequired,
-        }).isRequired,
-        data: PropTypes.array,
-        total: PropTypes.number,
+interface ChannelListProps {
+    actions: {
+        searchAllChannels: (term: string, notAssociatedToGroup?: string, excludeDefaultChannels?: boolean, page?: number, perPage?: number) => ActionFunc | ActionResult;
+        getData: (page: number, perPage: number, notAssociatedToGroup? : string, excludeDefaultChannels?: boolean) => ActionFunc | ActionResult;
+        removeGroup?: () => void;
     };
+    data?: {id: string; display_name: string}[];
+    total?: number;
+    removeGroup?: () => void;
+    onPageChangedCallback?: () => void;
+    emptyListTextId: string;
```

Comment:
About these props used by tests, I don't even think the tests are using them. We can probably remove them from here, and the tests.

Reply:
you are right

Reply:
Nevermind, they're passed down into `AbstractList`, my bad. Nothing to be done here.